envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -0,0 +1,155 @@
+syntax = "proto3";
+
+import "validate/validate.proto";
+
+package envoy.config.rbac.v2alpha;
+
+// RBAC (Role Based Access Control) defines ServiceRole and ServiceRoleBinding
+// objects.
+//
+// A ServiceRole specification includes a list of rules (permissions). Each rule has
+// the following standard fields:
+// * services: a list of services.
+// * verbs: HTTP verbs. In the case of gRPC, this field is ignored because the value is always
+// "POST".
+// * paths: HTTP paths or gRPC methods. Note that gRPC methods should be
+//   presented in the form of "packageName.serviceName/methodName".
+//
+// The fields used in ServiceRole definition map to attributes defined in AttributeContext
+// (//envoy/service/auth/v2/attribute_context.proto) as follows:
+//   service: destination.service
+//   method: request.http.method
+//   path: request.http.path
+//
+//  In addition, users can specify "constraints" on any attributes defined in AttributeContext
+//  (//envoy/service/auth/v2/attribute_context.proto).
+//
+// Below is an example of ServiceRole object "product-viewer", which has "read" ("GET" and "HEAD")
+// access to "products.default.svc.cluster.local" service at versions "v1" and "v2".
+//
+//     kind: ServiceRole
+//     metadata:
+//       name: products-viewer
+//       namespace: default
+//     spec:
+//       rules:
+//       - services: ["products.default.svc.cluster.local"]
+//         paths: ["/products/*"]
+//         verbs: ["GET", "HEAD"]
+//         constraints:
+//         - key: request.http.headers["version"]
+//           values: ["v1", "v2"]
+//
+// A ServiceRoleBinding specification includes two parts:
+// * "roleRef" refers to a ServiceRole object.
+// * A list of "subjects" that are assigned the roles.
+//
+// A subject can be a "user" or a "group", or represented with a set of attributes. Specifically,
+// "user" is mapped to "source.principal" attribute. "group" is currently undefined. It is reserved
+// to be used in the future.
+//
+// Below is an example of ServiceRoleBinding object "test-binding-products", which binds two
+// subjects to ServiceRole "product-viewer":
+//   * User (service account) "cluster.local/ns/default/sa/bookinfo-reviews"
+//   * Service "reviews.default.svc.cluster.local" at version "v1".
+//
+//     kind: ServiceRoleBinding
```

Comment:
Where does this kind/metadata stuff come from?

Reply:
This is the case that ServiceRole and ServiceRoleBinding expressed in K8S CRD format. I explained it in the new README.md.

"The storage of ServiceRole and ServiceRoleBinding objects are platform specific. For example,
 In [Istio](https://istio.io/docs/concepts/security/rbac.html), they are stored as [Kubernetes
 CustomResourceDefinition (CRD)](https://kubernetes.io/docs/concepts/api-extension/custom-resources/)
 objects. In the following document, we will see ServiceRole and ServiceRoleBinding examples
 expressed in Kubernetes CRD format."